tick/eval/node_types.go

```diff
@@ -0,0 +1,73 @@
+package eval
```

Comment:
This file is too isolated and it was confusing that the types were not defined next to their methods. I was not easy to learn the contents of the struct. Can we move the type definition right above each of the corresponding methods sets? 

That would leave only a single method in this file which seems like it should then be moved to `node_value_accessors.go` as well since its essentially a generic constructor.